docs/docs/development/new-behavior.mdx

```diff
@@ -390,135 +641,206 @@ static const struct behavior_driver_api behavior_bt_driver_api = {
+All functions that interact with layers can be found in `keymap.h`.
 
-### Defining Common Use-Cases for the Behavior (`.dtsi`) (Optional)
+Layers can be identified in two ways: a **order-independent** `zmk_keymap_layer_id_t`, and an `order-dependent` `zmk_keymap_layer_index_t`.
+Additionally, a bitmask, `zmk_keymap_layers_state_t`, is used to keep track of which layers are active or not, based on their **order-independent IDs**.
 
-`.dtsi` files, found in the directory `app/dts/behaviors/`, are only necessary for behaviors with more common use-cases. A common example is the mod-tap (`&mt`), which is a predefined type of hold-tap that takes a modifier key as the hold parameter and another key as the tap parameter.
+| Function                                                   | Description                                                                                                                  |
+| ---------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------- |
+| `zmk_keymap_layer_default(void)`                           | Returns a `zmk_keymap_layer_id_t` of the default layer.                                                                      |
+| `zmk_keymap_layer_state(void)`                             | Returns the current keyboard's layer state.                                                                                  |
+| `zmk_keymap_layer_active(zmk_keymap_layer_id_t layer)`     | Returns a `bool` representing if the layer with the chosen `zmk_keymap_layer_id_t` is active.                                |
+| `zmk_keymap_highest_layer_active(void)`                    | Returns a `zmk_keymap_layer_index_t` the ordered layer index of the highest active layer.                                    |
+| `zmk_keymap_layer_activate(zmk_keymap_layer_id_t layer)`   | Activates the chosen layer. Returns 0 if successful. Returns values < 0 if an error occurs.                                  |
+| `zmk_keymap_layer_deactivate(zmk_keymap_layer_id_t layer)` | Deactivates the chosen layer. Returns 0 if successful. Returns values < 0 if an error occurs.                                |
+| `zmk_keymap_layer_toggle(zmk_keymap_layer_id_t layer)`     | Toggles the chosen layer. Returns 0 if successful. Returns values < 0 if an error occurs.                                    |
+| `zmk_keymap_layer_to(zmk_keymap_layer_id_t layer)`         | Deactivates every layer, before activating the chosen layer. Returns 0 if successful. Returns values < 0 if an error occurs. |
+| `zmk_keymap_layer_name(zmk_keymap_layer_id_t layer)`       | Returns a C-string containing the layer's name.                                                                              |
```

Comment:
```suggestion
| `zmk_keymap_layer_name(zmk_keymap_layer_id_t layer)`       | Returns a C-string containing the layer's name.                                                                              |
| `zmk_keymap_layer_index_to_id(zmk_keymap_layer_index_t layer_index)` | Returns the order-independent ID for a given layer index.                                                                              |
```
I thought this is worth adding given the distinction made above, but feel free to reword